src/Scripts/azureHelper.ps1

```diff
@@ -75,25 +75,64 @@ function Azure-LoginServicePrincipal([string]$username, [string]$password, [stri
     }
 }
 
-function Azure-DeployZipToWebApp([string]$username, [string]$password, [string]$tenant, [string]$resourceGroup, [string]$name, [string]$zipPath)
+function Azure-DeployZipToWebApp([string]$username, [string]$password, [string]$tenant, [string]$resourceGroup, [string]$name, [string]$slot, [string]$zipPath)
 {
     Azure-LoginServicePrincipal $username $password $tenant
     
-    Write-Host "Deploying $name in resource group $resourceGroup"
+    if ($slot -eq $null -or $slot -eq "")
+    {
+        Write-Host "Deploying $name in resource group $resourceGroup into production slot"
 
-    & az webapp deploy --resource-group $resourceGroup --name $name --src-path $zipPath --type zip --async false
+        & az webapp deployment source config-zip --resource-group $resourceGroup --name $name --src $zipPath
```

Comment:
Could we not just use --slot "production" instead if it's null or empty?

Reply:
i thought that, but it doesn't work.
it says "there is no slot named production"
it would appear production is like some magic key word...